metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -1469,16 +1483,21 @@ def _get_request_json(self, url: str) -> Dict:
                 multiplier=self.config.api_options.retry_backoff_multiplier,
                 max=self.config.api_options.max_retry_interval,
             ),
-            retry=retry_if_exception_type(HTTPError429),
+            retry=retry_if_exception_type((HTTPError429, ConnectionError)),
             stop=stop_after_attempt(self.config.api_options.max_attempts),
         )
 
         @r.wraps
         def get_request():
             try:
-                response = self.session.get(url)
-                response.raise_for_status()
+                response = self.session.get(
+                    url, timeout=self.config.api_options.timeout
+                )
                 return response.json()
+            except ConnectionError as e:
```

Comment:
also if we have `retry_if_exception_type` for ConnectionError, then we should not be suppressing the ConnectionError exception

Reply:
For debug purpose i  have added this but not required removing this as we already added `retry_if_exception_type`